pandas/tseries/period.py

```diff
@@ -946,6 +946,27 @@ def repeat(self, n, *args, **kwargs):
         # overwrites method from DatetimeIndexOpsMixin
         return self._shallow_copy(self.values.repeat(n))
 
+    def where(self, cond, other=None):
+        """
+        .. versionadded:: 0.18.2
+
+        Return an Index of same shape as self and whose corresponding
+        entries are from self where cond is True and otherwise are from
+        other.
```

Comment:
its more complicated. I needed prelim logic different fro dti & period, so would have to have a helper function. We do this a fair amount, e.g. process for timezones, run a function, then reverse. Its ok the way we do it now, but it does duplicate *some* code.